rmgpy/molecule/fragment.py

```diff
@@ -123,6 +123,799 @@ def is_non_hydrogen(self):
+                label='',
+                species_repr=None,
+                vertices=None,
+                symmetry=-1,
+                multiplicity=-187,
+                reactive=True,
+                props=None,
+                inchi='',
+                smiles=''):
+        self.index = -1
+        self.label = label
+        self.species_repr = species_repr
+        Graph.__init__(self, vertices)
+        self.symmetry_number = symmetry
+        self._fingerprint = None
+        self._inchi = None
+        self._smiles = None
+        self.props = props or {}
+        self.multiplicity = multiplicity
+        self.reactive = reactive
+
+        if inchi and smiles:
+            logging.warning('Both InChI and SMILES provided for Fragment instantiation, '
+                            'using InChI and ignoring SMILES.')
+        if inchi:
+            self.from_inchi(inchi)
+            self._inchi = inchi
+        elif smiles:
+            self.from_smiles_like_string(smiles)
+            self._smiles = smiles
+
+    def __deepcopy__(self, memo):
+        return self.copy(deep=True)
+
+    def __str__(self):
+        """
+        Return a string representation, in the form 'label(id)'.
+        """
+        if self.index == -1: return self.label
+        else: return '{0}({1:d})'.format(self.label, self.index)
+
+    def __getAtoms(self): return self.vertices
+    def __setAtoms(self, atoms): self.vertices = atoms
+    atoms = property(__getAtoms, __setAtoms)
+
+    def draw(self, path):
+        """
+        Generate a pictorial representation of the chemical graph using the
+        :mod:`draw` module. Use `path` to specify the file to save
+        the generated image to; the image type is automatically determined by
+        extension. Valid extensions are ``.png``, ``.svg``, ``.pdf``, and
+        ``.ps``; of these, the first is a raster format and the remainder are
+        vector formats.
+        """
+        from rmgpy.molecule.draw import MoleculeDrawer
+        format = os.path.splitext(path)[-1][1:].lower()
+        MoleculeDrawer().draw(self, format, target=path)
+
+    def _repr_png_(self):
+        """
+        Return a png picture of the molecule, useful for ipython-qtconsole.
+        """
+        from rmgpy.molecule.draw import MoleculeDrawer
+        tempFileName = 'temp_molecule.png'
+        MoleculeDrawer().draw(self, 'png', tempFileName)
+        png = open(tempFileName,'rb').read()
+        os.unlink(tempFileName)
+        return png
+
+    def copy(self, deep=False):
+        """
+        Create a copy of the current graph. If `deep` is ``True``, a deep copy
+        is made: copies of the vertices and edges are used in the new graph.
+        If `deep` is ``False`` or not specified, a shallow copy is made: the
+        original vertices and edges are used in the new graph.
+        """
+        g = Graph.copy(self, deep)
+        other = Fragment(vertices=g.vertices)
+        return other
```

Comment:
Do the other attributes of the Fragment object need to be copied?

Reply:
Just modified it to copy connectivity and sorting_label as well!